JavadocStyleGuide.md

```diff
@@ -0,0 +1,223 @@
+# Gradle Javadoc Style Guide
+
+## 1.1 Formatting
+
+### 1.1.1 General form
+
+The basic formatting of Javadoc blocks is as seen in this example:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ * The url argument must specify an absolute <a href="#{@link}">{@link URL}</a>. The name
+ * argument is a specifier that is relative to the url argument.
+ * <p>
+ * This method always returns immediately, whether or not the
+ * image exists. When this applet attempts to draw the image on
+ * the screen, the data will be loaded. The graphics primitives
+ * that draw the image will incrementally paint on the screen.
+ *
+ * @param  url  an absolute URL giving the base location of the image
+ * @param  name the location of the image, relative to the url argument
+ * @return      the image at the specified URL
+ * @see         Image
+ */
+public Image getImage(URL url, String name) {
+   try {
+       return getImage(new URL(url, name));
+   } catch (MalformedURLException e) {
+       return null;
+   }
+}
+```
+
+Basic formatting rules:
+
+- The first line contains the begin-comment delimiter ( `/**`).
+- Notice the inline tag `{@link URL}`, which converts to an HTML hyperlink pointing to the documentation for the URL class.
+- If you have more than one paragraph in the doc comment, separate the paragraphs with a `<p>` paragraph tag, as shown.
```

Comment:
When I type `<p>` in a javadoc segment, IntelliJ automatically adds the corresponding `</p>` for me. For this reason, I have been surrounding each paragraph with `<p>text</p>`. 

I understand the official Oracle javadoc style guide says to use a single `<p>` between paragraphs, so if we choose to enforce this bullet, we should try to fix IntellIJ from adding `</p>` automatically. Otherwise, the IDE suggestions directly conflict with this style rule.

Reply:
It doesn't do this for me...don't know how I configured this though.

Reply:
Settings -> Editor -> General -> Smart Keys -> "Insert closing tag on tag completion"